miio/cooker.py

```diff
@@ -0,0 +1,706 @@
+        """
+        self.data = [int(data[i:i + 2], 16) for i in range(0, len(data), 2)]
+
+    @property
+    def temperatures(self) -> list[int]:
+        return self.data
+
+    def __str__(self) -> str:
+        return "".join(["{:02x}".format(value) for value in self.data])
+
+    def __repr__(self) -> str:
+        s = "<TemperatureHistory temperatures=%s>" % self.data
+        return s
+
+
+class CookerCustomizations:
+    def __init__(self, custom: str):
+        """
+        Example values:
+
+        ffffffffffff011effff010000001d1f,
+        ffffffffffff011effff010004026460,
+        ffffffffffff011effff01000a015559,
+        ffffffffffff011effff01000000535d
+        """
+        self.custom = [int(custom[i:i + 2], 16) for i in
+                       range(0, len(custom), 2)]
+
+    @property
+    def jingzhu_appointment(self) -> time:
+        return time(hour=self.custom[0], minute=self.custom[1])
+
+    @property
+    def kuaizhu_appointment(self) -> time:
+        return time(hour=self.custom[2], minute=self.custom[3])
+
+    @property
+    def zhuzhou_appointment(self) -> time:
+        return time(hour=self.custom[4], minute=self.custom[5])
+
+    @property
+    def zhuzhou_cooking(self) -> time:
+        return time(hour=self.custom[6], minute=self.custom[7])
+
+    @property
+    def favorite_appointment(self) -> time:
+        return time(hour=self.custom[8], minute=self.custom[9])
+
+    @property
+    def favorite_cooking(self) -> time:
+        return time(hour=self.custom[10], minute=self.custom[11])
+
+    def __str__(self) -> str:
+        return "".join(["{:02x}".format(value) for value in self.custom])
+
+    def __repr__(self) -> str:
+        s = "<CookerCustomizations jingzhu_appointment=%s, " \
+            "kuaizhu_appointment=%s, " \
+            "zhuzhou_appointment=%s, " \
+            "zhuzhou_cooking=%s, " \
+            "favorite_appointment=%s," \
+            "favorite_cooking=%s>" % \
+            (self.jingzhu_appointment,
+             self.kuaizhu_appointment,
+             self.zhuzhou_appointment,
+             self.zhuzhou_cooking,
+             self.favorite_appointment,
+             self.favorite_cooking)
+        return s
+
+
+class CookingStage:
+    def __init__(self, stage: str):
+        """
+        Example timeouts: 'null', 02000000ff, 03000000ff, 0a000000ff, 1000000000
+
+        The meaning of stage[8:10] is unknown.
+        """
+        self.stage = stage
+
+    @property
+    def state(self) -> int:
+        """
+
+        10: Cooking finished
+        11: Cooking finished
+        12: Cooking finished
+        """
+        return int(self.stage[0:2], 16)
+
+    @property
+    def rice_id(self) -> int:
+        return int(self.stage[2:6], 16)
+
+    @property
+    def taste(self) -> int:
+        return int(self.stage[6:8], 16)
+
+    @property
+    def taste_phase(self) -> int:
+        phase = self.taste / 33
+
+        if phase > 2:
+            return 2
```

Comment:
This returns always either None or 2, is that correct? If yes, I'd say make returning None explicit + add `Optional` to return value.

Reply:
Good catch! Fixed.